### PR TITLE
Auto Isolation

### PR DESCRIPTION
**Pull request by bkendall**
**Warning: this is very much an alpha feature**

Adds options to control auto-isolation. This has been used by @prafulrana for a while and been tested by him.
### Dependencies
- [x] Runnable/api-client#12
  - [x] update reference in `package.json`
### Reviewers
- [x] anand

### CODE REVIEW

File: lib/auto-isolation.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+const find = require('101/find')
+
+const Runnable = require('./runnable')
+const Utils = require('./utils')
+
+const read = Promise.promisify(require('read'))
+
+class AutoIsolation extends Runnable {
+  static handleAutoIsolation (args) {
+    if (!args.enable && !args.disable) {
```
Comment (anandkumarpatel) [inline lib/auto-isolation.js:17]: can also be `!(args.enable || args.disable)`

File: lib/auto-isolation.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      return AutoIsolation.list(args)
+    }
+    return Utils.getRepositoryAndMasterPodInstance(args)
+      .spread((argsWithRepo, instance) => {
+        const lowerRepo = argsWithRepo.repository.toLowerCase()
```
Comment (anandkumarpatel) [inline lib/auto-isolation.js:22]: any chance `argsWithRepo` will be null?

File: lib/auto-isolation.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        qs: { instance: args._instance._id }
+      }
+      AutoIsolation.user.fetchAutoIsolationConfigs(opts, callback)
+    })
+      .then((configs) => {
+        if (!configs.length) {
+          console.log(`No configurations exist for ${args.repository}.`)
+        }
+        return Promise.fromCallback((callback) => {
+          AutoIsolation.user.newAutoIsolationConfig(configs[0]._id).destroy(callback)
```
Comment (anandkumarpatel) [inline lib/auto-isolation.js:51]: hmm hardcoded config to `0` here? is there a reason why configs is an array then?

File: lib/auto-isolation.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                '',
+                '(`q` to stop) >'
+              ].join('\n')
+              return read({ prompt: prompt })
+                .then((selection) => {
+                  if (selection.toLowerCase() === 'q' || selection === '') {
+                    done = true
+                    return
+                  }
+                  const index = parseInt(selection, 10) - 1
```
Comment (anandkumarpatel) [inline lib/auto-isolation.js]: does this validate `selection` is a number? maybe add a try catch?

**Comment by anandkumarpatel** [comment]
looking good, just have a few questions for my own understanding

File: lib/auto-isolation.js (hunk 1 above)
Comment (bkendall) [inline lib/auto-isolation.js:17]: true, mr. logic!

File: lib/auto-isolation.js (hunk 2 above)
Comment (bkendall) [inline lib/auto-isolation.js:22]: no

File: lib/auto-isolation.js (hunk 3 above)
Comment (bkendall) [inline lib/auto-isolation.js:51]: yes. it's always an array when it comes back

Comment (bkendall) [inline lib/auto-isolation.js:51]: (it fetched them - fetches are always arrays)

File: lib/auto-isolation.js (hunk 4 above)
Comment (bkendall) [inline lib/auto-isolation.js]: nope! I should, you're right

Comment (bkendall) [inline lib/auto-isolation.js]: (also, try/catch is unnecessary, just need to check if `NaN` was returned)

**Pull request merged**